File: backend/llm/llm.go
package llm

import (
	"context"
)

func QueryLLM(ctx context.Context, input, systemContent string, modelName string, temperature float32, maxOutputTokens int32) (res string, err error) {
	// return QueryGemini(ctx, input, systemContent, "gemini-1.5-pro-002", temperature, maxOutputTokens)
	return querySambaNova(ctx, input, systemContent, modelName, temperature)
	//switch modelName {
	//case "doubao":
	//	return queryVolEngine(ctx, input, systemContent, 0.98)
	//case "gemini-1.5-pro-002":
	//	return QueryGemini(ctx, input, systemContent, modelName, temperature, maxOutputTokens)
	//}
	//return "", fmt.Errorf("wrong llm")
}

var transLateSys = `把输入完全翻译成英文，不要输出翻译文本以外的内容，只需要输出翻译后的文本。
如果包含翻译之外的内容，则重新输出`

func LLMTranslate(ctx context.Context, input string) (output string, err error) {
	return querySiliconFlow(ctx, input, transLateSys, 0.01)
}

File: backend/llm/llm.py
from backend.llm.openai import query_openai
from backend.llm.sambanova import query_samba_nova
from backend.llm.siliconflow import query_silicon_flow
from backend.util.file import get_config

def query_llm(input_text: str, system_content: str, model_name: str, temperature: float, max_output_tokens: int) -> str:
    url = get_config()['url']
    if "sambanova" in url.lower():
        return query_samba_nova(input_text, system_content, model_name, temperature)
    return query_openai(input_text, system_content, model_name, temperature)

def llm_translate(input_text: str) -> str:
    translate_sys = "把输入完全翻译成英文，不要输出翻译文本以外的内容，只需要输出翻译后的文本。如果包含翻译之外的内容，则重新输出"
    return query_silicon_flow(input_text, translate_sys, 0.01)


# Example usage:
# result, error = llm_translate("你好")
# if error:
#     print(f"Error: {error}")
# else:
#     print(result)